test/extensions/filters/network/mongo_proxy/config_test.cc

```diff
@@ -67,186 +70,135 @@ TEST(MongoFilterConfigTest, MongoFilterWithEmptyProto) {
-      "stat_prefix": "my_stat_prefix",
-      "fault" : {
-        "fixed_delay": {
-          "percent": 101,
-          "duration_ms": 1
-        }
-      }
-    }
+    const std::string yaml_string = R"EOF(
+    stat_prefix: my_stat_prefix
+    delay:
+      percentage:
+        numerator: -1
+        denominator: HUNDRED
+      fixed_delay: 1s
     )EOF";
 
-    handleInvalidConfiguration(json_string);
-  }
-
-  {
-    std::string json_string = R"EOF(
-    {
-      "stat_prefix": "my_stat_prefix",
-      "fault" : {
-        "fixed_delay": {
-          "percent": -1,
-          "duration_ms": 1
-        }
-      }
-    }
-    )EOF";
-
-    handleInvalidConfiguration(json_string);
+    handleInvalidConfiguration(yaml_string, R"(invalid value -1 for type TYPE_UINT32)");
   }
 }
 
 TEST(MongoFilterConfigTest, InvalidFaultsType) {
   {
-    std::string json_string = R"EOF(
-    {
-      "stat_prefix": "my_stat_prefix",
-      "fault" : {
-        "fixed_delay": {
-          "percent": "df",
-          "duration_ms": 1
-        }
-      }
-    }
+    const std::string yaml_string = R"EOF(
+    stat_prefix: my_stat_prefix
+    delay:
+      percentage:
+        numerator: df
+        denominator: HUNDRED
+      fixed_delay: 1s
     )EOF";
 
-    handleInvalidConfiguration(json_string);
+    handleInvalidConfiguration(yaml_string, R"(invalid value "df" for type TYPE_UINT32)");
   }
 
   {
-    std::string json_string = R"EOF(
-    {
-      "stat_prefix": "my_stat_prefix",
-      "fault" : {
-        "fixed_delay": {
-          "percent": 3,
-          "duration_ms": "ab"
-        }
-      }
-    }
+    const std::string yaml_string = R"EOF(
+    stat_prefix: my_stat_prefix
+    delay:
+      percentage:
+        numerator: 1
+        denominator: HUNDRED
+      fixed_delay: ab
     )EOF";
 
-    handleInvalidConfiguration(json_string);
+    handleInvalidConfiguration(yaml_string, "Illegal duration format; duration must end with 's'");
   }
 
   {
-    std::string json_string = R"EOF(
-    {
-      "stat_prefix": "my_stat_prefix",
-      "fault" : {
-        "fixed_delay": {
-          "percent": 3,
-          "duration_ms": "0"
-        }
-      }
-    }
+    const std::string yaml_string = R"EOF(
+    stat_prefix: my_stat_prefix
+    delay:
+      percentage:
+        numerator: 3
+        denominator: HUNDRED
+      fixed_delay: 0s
     )EOF";
 
-    handleInvalidConfiguration(json_string);
+    handleInvalidConfiguration(yaml_string,
+                               R"(FixedDelay: \["value must be greater than " "0s"\])");
   }
 }
 
 TEST(MongoFilterConfigTest, CorrectFaultConfiguration) {
-  std::string json_string = R"EOF(
-  {
-    "stat_prefix": "my_stat_prefix",
-    "fault" : {
-      "fixed_delay": {
-        "percent": 1,
-        "duration_ms": 1
-      }
-    }
-  }
+  const std::string yaml_string = R"EOF(
+  stat_prefix: my_stat_prefix
+  delay:
+    percentage:
+      numerator: 1
+      denominator: HUNDRED
+    fixed_delay: 1s
```

Comment:
These numbers looks different, i.e before it was 1ms duration, now it's 1s.

Reply:
yep you're correct. They don't affect the test in this case but I'll preserve.